src/commands/dao_cmds/install.js

```diff
@@ -38,72 +43,119 @@ exports.builder = function (yargs) {
-          getContract('@aragon/os', 'Kernel').abi,
-          dao
+        const initPayload = encodeInitPayload(web3, ctx.repo.abi, appInit, appInitArgs)
+
+        if (initPayload === '0x') {
+          ctx.notInitialized = true
+        }
+
+        const getTransactionPath = (wrapper) => {
+          const fnArgs = [
+            ctx.repo.appId,
+            ctx.repo.contractAddress,
+            initPayload,
+            false
+          ]
+          return wrapper.getTransactionPath(dao, 'newAppInstance', fnArgs)
+        }
+
+        return execTask(dao, getTransactionPath, { reporter, apm: apmOptions, web3, wsProvider })
+      }
+    },
+    {
+      title: 'Fetching deployed app',
+      enabled: () => setPermissions,
+      task: async (ctx, task) => {
+        const logABI = kernelABI.find(
+          ({ type, name }) => type === 'event' && name === 'NewAppProxy'
         )
 
-        ctx.aclAddress = await kernel.methods.acl().call()
-        if (!ctx.accounts) {
-          ctx.accounts = await web3.eth.getAccounts()
+        if (!logABI) {
+          throw new Error('Kernel ABI in aragon.js doesnt contain NewAppProxy log')
         }
 
-        // TODO: report if empty
-        const initPayload = encodeInitPayload(web3, ctx.repo.abi, appInit, appInitArgs)
+        const logSignature = `${logABI.name}(${logABI.inputs.map(i => i.type).join(',')})`
+        const logTopic = web3.utils.sha3(logSignature)
+        const deployLog = ctx.receipt.logs.find(
+          ({ topics, address }) => {
+            return topics[0] === logTopic && addressesEqual(dao, address)
+          }
+        )
 
-        if (initPayload === '0x') {
-          ctx.notInitialized = true
+        if (!deployLog) {
+          task.skip('App wasn\'t deployed in transaction.')
+          return
         }
 
-        const { events } = await kernel.methods.newAppInstance(
-          ctx.repo.appId,
-          ctx.repo.contractAddress,
-          initPayload,
-          false
-        ).send({
-          from: ctx.accounts[0],
-          gasLimit: 1e6
-        })
-
-        ctx.appAddress = events['NewAppProxy'].returnValues.proxy
+        const log = web3.eth.abi.decodeLog(logABI.inputs, deployLog.data)
+        ctx.appAddress = log.proxy
       }
     },
     {
       title: 'Set permissions',
+      enabled: (ctx) => setPermissions && ctx.appAddress,
       task: async (ctx, task) => {
+        const aclAddress = await kernel.methods.acl().call()
+
         if (!ctx.repo.roles || ctx.repo.roles.length === 0) {
           throw new Error('You have no permissions defined in your arapp.json\nThis is required for your app to properly show up.')
         }
 
         const permissions = ctx.repo.roles
           .map((role) => [ANY_ENTITY, ctx.appAddress, role.bytes])
 
-        return setPermissions(
+        // TODO: setPermissions should use ACL functions with transaction pathing
+        return setPermissionsWithoutTransactionPathing(
```

Comment:
`dao install` still doesn't set permissions using transaction pathing, so this will fail when the sender cannot set the permission directly. I think it should be its own issue though